### ISSUE TITLE
ThemeData.brightness is not correctly propagated to Cupertino widgets

### ISSUE THREAD

**Issue by pedromassango** [flutter/flutter#47255]
After the upgrade to Flutter 1.12.xx I discovered that the CupertinoActivityIndicator of my app turned dark.

## Steps to Reproduce

1. Have this in my `MaterialApp` :

```dart
        theme: ThemeData(
          primarySwatch: Colors.deepOrange,
          accentColor: MyColors.accentColor,
          scaffoldBackgroundColor: Color(0xFF1E1D25), // Dark color
          fontFamily: 'Roboto',
          brightness: Brightness.dark, // Dark theme
          primaryColorBrightness: Brightness.dark, // Dark brightness
          bottomSheetTheme: BottomSheetThemeData(
            backgroundColor: MyColors.transparent
          )
        )
```

With this configurations the `CupertinoActivityIndicator` still with the dark color (almost invisible in my case). Can you confirm if my `ThemeData` is well configured?

2. Show a `CupertinoActivityIndicator` in a Flutter page
3. Note that the indicator have the dark color (almost invisible)

**Target Platform:** iOS
**Devices:** iPhone 6s

## Logs
```
[√] Flutter (Channel stable, v1.12.13+hotfix.5, on Microsoft Windows [Version 10.0.18362.535], locale en-US)
    • Flutter version 1.12.13+hotfix.5 at C:\src\flutter
    • Framework revision 27321ebbad (7 days ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0


[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\src\android
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 28.0.3
    • ANDROID_HOME = C:\src\android
    • Java binary at: C:\Users\Pedro Massango\Downloads\android-studio-ide-192.5959023-windows\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[!] Android Studio (version 4.0)
    • Android Studio at C:\Users\Pedro Massango\Downloads\android-studio-ide-192.5959023-windows\android-studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] IntelliJ IDEA Community Edition (version 2019.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2019.1.1
    • Flutter plugin version 42.1.4
    • Dart plugin version 193.5731

[√] VS Code (version 1.40.1)
    • VS Code at C:\Users\Pedro Massango\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.6.0

[√] Connected device (1 available)
    • Redmi 5 Plus • 0258ff700005 • android-arm64 • Android 8.1.0 (API 27)

! Doctor found issues in 1 category.

```

**Comment by LongCatIsLooong** [comment]
`CupertinoActivityIndicator` is working as expected. The real issue is `Theme` failed to propagate its brightness to the Cupertino world so all Cupertino widgets are still following the brightness specified by the operating system.

I've opened a pull request to fix this but since it's a breaking change it may take slightly longer to land in master. In the meantime, as a workaround you can add this line to your `ThemeData` to force Cupertino widgets into dark mode.
```dart
cupertinoOverrideTheme: const CupertinoThemeData(brightness: Brightness.dark),
```

**Comment by pedromassango** [comment]
Thank you

**Comment by pushangupta** [comment]
Happens in material widgets too

**Comment by LongCatIsLooong** [comment]
@pushangupta sorry somehow I let the comment slip through. It is very likely the problem you're seeing is caused by something else. Could you open a new issue and add more details to help investigate?

**Comment by tuanbs** [comment]
> `CupertinoActivityIndicator` is working as expected. The real issue is `Theme` failed to propagate its brightness to the Cupertino world so all Cupertino widgets are still following the brightness specified by the operating system.
> 
> I've opened a pull request to fix this but since it's a breaking change it may take slightly longer to land in master. In the meantime, as a workaround you can add this line to your `ThemeData` to force Cupertino widgets into dark mode.
> 
> ```dart
> cupertinoOverrideTheme: const CupertinoThemeData(brightness: Brightness.dark),
> ```

I see this issue is fixed in `Flutter version `1.12.13+hotfix.9`. Thanks guys.
